### ISSUE TITLE
`getent passwd` shows root user twice

### ISSUE THREAD

**Issue by jlebon** [systemd/systemd#15160]
**systemd version the issue has been seen with**

```
# rpm -q systemd
systemd-245~rc1-4.fc32.x86_64
```

**Used distribution**

Fedora 32

**Expected behaviour you didn't see**

`root` user shows up once in the output of `getent passwd`

**Unexpected behaviour you saw**

`root` user shows up twice in the output of `getent passwd`:

```
# grep ^root /etc/passwd
root:x:0:0:root:/root:/bin/bash
# getent passwd | grep ^root
root:x:0:0:root:/root:/bin/bash
root:x:0:0:root:/root:/bin/sh
```

**Steps to reproduce the problem**

1. boot Fedora cloud image
2. run `getent passwd`

I actually first saw this in Fedora CoreOS 32 (https://github.com/coreos/fedora-coreos-config/pull/306), but it doesn't seem FCOS-specific.

I think this is some kind of mishandling of the root intrinsic user if it already exists in `/etc/passwd`? I initially thought this was somehow related to the systemd NSS plugin getting confused by the root shell defaulting to `/bin/sh` while the `/etc/passwd` entry uses `/bin/bash`, but changing the root shell to `/bin/sh` in `/etc/passwd` will still show two `root` entries. (Though `userdbctl -j` does show only a single root user, with `/bin/bash` as its shell).

**Comment by mbiebl** [comment]
I can confirm that (on Debian, where we have userdb disabled):

```
$ getent passwd | grep ^root
root:x:0:0:root:/root:/bin/bash
root:x:0:0:root:/root:/bin/sh
```


I also see a duplicated nobody account:

```
$ getent passwd | grep nobody
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
nobody:x:65534:65534:nobody:/:/usr/sbin/nologin
```

**Comment by jlebon** [comment]
I was surprised to encounter this somewhat significant change in f32. OOC, was enabling `systemd-userdbd` by default discussed anywhere? It's not in https://fedoraproject.org/wiki/Releases/32/ChangeSet at least and I couldn't find mentions of it on the devel list.

I'd be interested to know why Debian is disabling this and whether those reasons also apply to Fedora.

**Comment by marcthe12** [comment]
Ran a test on arch Linux. Since arch uses sysuser.d for everything except root.

I deleted the nobody user.
I also commented the nobody user from basic.conf from systemd. Which solves the issues.

For root it's a bit complicated as it needs the ability to be configured since there might be a need to login as root.

Maybe we can start by deleting the intrinsic users from basic.conf.

**Comment by mbiebl** [comment]
The problem is that systemd synthesizes a second root and nobody user. This needs to be fixed.

**Comment by marcthe12** [comment]
@mbiebl Agreed. Just wanted to ask since root and nobody user is also created by sysuser on first boot if missing from /etc/passwd, what is the usecase of the intrinsic users and is there a need to still define them in basic.conf

**Comment by poettering** [comment]
Fix waiting in #15553
